test/common/http/conn_manager_impl_test.cc

```diff
@@ -199,8 +199,14 @@ class HttpConnectionManagerImplTest : public Test, public ConnectionManagerConfi
       EXPECT_CALL(*filter, onDestroy());
     }
 
-    for (auto filter : encoder_filters_) {
-      EXPECT_CALL(*filter, onDestroy());
+    if (reverse_encoders_order_) {
+      for (int i = encoder_filters_.size() - 1; i >= 0; --i) {
```

Comment:
This might be clearer using `std::for_each` with reversed and regular iterators. Then you can do something like:

```
auto setup_filter_expect = [](MockStreamEncoderFilter* filter){ EXPECT_CALL(*filter, onDestroy()); };
if (reverse_encoders_order_) {
  std::for_each(encoder_filters.rbegin(), encoder_filters.rend(), setup_filter_expect);
} else {
  std::for_each(encoder_filters.begin(), encoder_filters.end(), setup_filter_expect);
```

This should let you write the same style code for reverse and forward, which seems to have some readability benefit.

Reply:
@dnoe good idea! Incorporated your suggestion.